src/plugins/demo/appAdapter.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+
+import type {PluginDeclaration} from "../../analysis/pluginDeclaration";
+import {declaration} from "./declaration";
+import type {
+  StaticAppAdapter,
+  DynamicAppAdapter,
+} from "../../app/adapters/appAdapter";
+import {StaticAdapterSet} from "../../app/adapters/adapterSet";
+import {Assets} from "../../app/assets";
+import {type RepoId, makeRepoId} from "../../core/repoId";
+import {NodeAddress, type NodeAddressT} from "../../core/graph";
+import {graph} from "./graph";
+
+export class FactorioStaticAdapter implements StaticAppAdapter {
+  loadingMock: Function;
+  declaration(): PluginDeclaration {
+    return declaration;
+  }
+  async load(assets: Assets, repoId: RepoId): Promise<FactorioDynamicAdapter> {
+    const result: FactorioDynamicAdapter = new FactorioDynamicAdapter();
+    if (this.loadingMock) {
+      return this.loadingMock(assets, repoId).then(() => result);
+    }
+    return Promise.resolve(result);
+  }
+}
+
+export class FactorioDynamicAdapter implements DynamicAppAdapter {
+  graph() {
+    return graph();
+  }
+  nodeDescription(x: NodeAddressT) {
+    return NodeAddress.toString(x);
```

Comment:
Can this prepend `"[Factorio] "` or something, so that we can tell in
test code where this is being called versus a `toString` fallback?